src/ActionList/Item.tsx

```diff
@@ -212,13 +209,16 @@ export const Item = React.forwardRef<HTMLLIElement, ActionListItemProps>(
       id: itemId,
     }
 
-    const containerProps = _PrivateItemWrapper ? {role: role || itemRole ? 'none' : undefined} : menuItemProps
-
+    // If there is a private item wrapper, menu props are passed to the ItemWrapper, otherwise to the outer container
+    const containerProps = _PrivateItemWrapper ? {} : menuItemProps
     const wrapperProps = _PrivateItemWrapper ? menuItemProps : {}
 
     return (
       <ItemContext.Provider value={{variant, disabled, inlineDescriptionId, blockDescriptionId}}>
-        <LiBox
+        <Box
+          data-component="ActionList.Item--Container"
+          // If role is specified on the item, or gathered from the list, we render the wrapper as a div, otherwise as a li
+          as={role || itemRole ? undefined : 'li'}
```

Comment:
While it does a bit icky that we are overriding the ul semantics when we use `<ul role="listbox">`, we don't have to go out of our way to make it a div either 😅 

(if using a div or ul is all the same) I would recommend picking the element that leads to the simplest implementation. 

Related: Does the type for forwardedRef also change based on role? If yes, that feels like an implementation detail that is leaking out (especially when it's not explicitly set by the developer using the component, but internally inside an ActionMenu). Would it help to make that simpler by always rendering `ul`?